source/core_http_client.c

```diff
@@ -1378,18 +1360,18 @@ static HTTPStatus_t addHeader( HTTPRequestHeaders_t * pRequestHeaders,
     assert( fieldLen != 0U );
     assert( valueLen != 0U );
 
-    pBufferCur = ( char * ) ( pRequestHeaders->pBuffer + pRequestHeaders->headersLen );
+    pBufferCur = ( char * ) &( pRequestHeaders->pBuffer[ pRequestHeaders->headersLen ] );
     backtrackHeaderLen = pRequestHeaders->headersLen;
 
     /* Backtrack before trailing "\r\n" (HTTP header end) if it's already written.
      * Note that this method also writes trailing "\r\n" before returning.
      * The first condition prevents reading before start of the header. */
     if( ( HTTP_HEADER_END_INDICATOR_LEN <= pRequestHeaders->headersLen ) &&
-        ( strncmp( ( char * ) pBufferCur - HTTP_HEADER_END_INDICATOR_LEN,
+        ( strncmp( ( char * ) &pBufferCur[ 0 - ( ( int ) HTTP_HEADER_END_INDICATOR_LEN ) ],
```

Comment:
Don't negative array indexes require C99?
The [CMakeLists.txt file in the test directory](https://github.com/FreeRTOS/coreHTTP/blob/main/test/CMakeLists.txt#L10) still claims C90:
```
# Use C90.
set( CMAKE_C_STANDARD 90 )
set( CMAKE_C_STANDARD_REQUIRED ON )
```

Reply:
Updated CMake to use C99 as lhttp requires use of C99.

Reply:
This isnt the standard that is required; it is the standard that the compilation objects are compiled with. If coreHTTP should be compiled with `-std=c89` then that should remain CMAKE_C_STANDARD 90. It will still link fine against llhttp built with `-std=c99`.

Reply:
We are using this for running unit-tests so I think that should be fine.